lib/src/web/database/web_db_context.dart

```diff
@@ -0,0 +1,98 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+import 'package:sqlite_async/sqlite3_common.dart';
+import 'package:sqlite_async/src/sqlite_connection.dart';
+import 'executor/sqlite_executor.dart';
+
+/// Custom function which exposes CommonDatabase.autocommit
+const sqliteAsyncAutoCommitCommand = 'sqlite_async_autocommit';
+
+class WebReadContext implements SqliteReadContext {
+  SQLExecutor db;
+  bool _closed = false;
+
+  @protected
+  bool isTransaction;
+
+  WebReadContext(this.db, {this.isTransaction = false});
+
+  @override
+  Future<T> computeWithDatabase<T>(
+      Future<T> Function(CommonDatabase db) compute) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<Row> get(String sql, [List<Object?> parameters = const []]) async {
+    return (await getAll(sql, parameters)).first;
+  }
+
+  @override
+  Future<ResultSet> getAll(String sql,
+      [List<Object?> parameters = const []]) async {
+    if (_closed) {
+      throw SqliteException(0, 'Transaction closed', null, sql);
+    }
+    return db.select(sql, parameters);
+  }
+
+  @override
+  Future<Row?> getOptional(String sql,
+      [List<Object?> parameters = const []]) async {
+    try {
+      return (await getAll(sql, parameters)).first;
+    } catch (ex) {
+      return null;
```

Comment:
There could be exceptions other than "no results" caught here. Rather use an implementation like this:

https://github.com/powersync-ja/sqlite_async.dart/blob/fdd09c34475dbd5aa6681c4d3af1dadb5c9732b8/lib/src/native/database/native_sqlite_connection_impl.dart#L226